server/sonar-server/src/main/java/org/sonar/server/computation/qualitymodel/RatingSettings.java

```diff
@@ -50,7 +51,11 @@ public RatingSettings(Settings settings) {
     String[] languageConfigIndexes = settings.getStringArray(LANGUAGE_SPECIFIC_PARAMETERS);
     for (String languageConfigIndex : languageConfigIndexes) {
       String languagePropertyKey = LANGUAGE_SPECIFIC_PARAMETERS + "." + languageConfigIndex + "." + LANGUAGE_SPECIFIC_PARAMETERS_LANGUAGE_KEY;
-      builder.put(settings.getString(languagePropertyKey), LanguageSpecificConfiguration.create(settings, languageConfigIndex));
+      String languageKey = settings.getString(languagePropertyKey);
+      if (languageKey == null) {
+        throw MessageException.of("Technical debt configuration is corrupted. At least one Language specific parameter has no language key");
```

Comment:
I think it deserves to give hints about where this is configured (General Settings -> Technical Debt -> property "Language specific parameters")